src/future/poll_fn.rs

```diff
@@ -23,15 +23,18 @@ use crate::task::{Context, Poll};
 /// #
 /// # })
 /// ```
-pub async fn poll_fn<F, T>(f: F) -> T
+pub fn poll_fn<F, T>(f: F) -> PollFn<F>
```

Comment:
Hmm, I'm so-so about this change. Functions such as [`io::copy`](https://docs.rs/async-std/1.5.0/async_std/io/fn.copy.html) don't work on no-std either, but would be really useful to have as well. Having these functions be `async fn` was a bit of a deliberate choice when designing async-std.

With [async await for embedded rust](https://ferrous-systems.com/blog/async-on-embedded/) steadily progressing, I wonder how needed this is. It seems as async/await continues to move beyond mvp status this will be much of a non-issue.

All in all I think it'd be okay to leave this method as-is for now, and we can merge all the other changes. Thanks!

Reply:
Hm, the feature won't make the cut in 6 days, so it's at least 18 weeks out. Given that this is no user-visible change, I'd be fine with enabling embrio users to use this currently.

Reply:
Since the interface seen from the user does not change, I think that it is OK to change this.